sdk/src/connections/target/addi9036_sensor.cpp

```diff
@@ -620,61 +620,123 @@ aditof::Status Addi9036Sensor::getFrame(uint16_t *buffer,
             return bytesused == (width * height * 3);
         };
 
-        if (width == 668) {
-            unsigned int j = 0;
-            for (unsigned int i = 0; i < (buf_data_len); i += 3) {
-                if ((i != 0) && (i % (336 * 3) == 0)) {
-                    j -= 4;
-                }
+#if defined(JETSON)
 
-                buffer[j] = (((unsigned short)*(pdata[0] + i)) << 4) |
-                            (((unsigned short)*(pdata[0] + i + 2)) & 0x000F);
-                j++;
+        if (m_implData->frameDetails.type == "depth") {
+            memcpy(buffer, pdata[0], buf[0].bytesused);
+        } else if (m_implData->frameDetails.type == "ir") {
+
+            memcpy(buffer, pdata[0], buf[0].bytesused);
```

Comment:
this line is exactly as the one at line 626. The code would be more readable if you merge the 2 if branches